### PR TITLE
Better promises

### PR DESCRIPTION
**Pull request by jsturtevant**
Cleaning up the code to make it more readable/maintainable.

### CODE REVIEW

File: EvaluateAutoScale/index.js [hunk 1]
```diff
@@ -8,84 +8,54 @@ module.exports = function (context, req) {
...
         context.done();
     });
 
 };
 
-function evaluateAutoScale(batch_client, poolid, maxNodes){
+function isAutoScaleEnabled(poolInfo){
+    if(poolInfo.state != "active")
+    {
+        Promise.reject({code: "notActive"});
```
Comment (noopkat) [inline EvaluateAutoScale/index.js]: this should probably return the rejection, as it's the static Promise use

File: EvaluateAutoScale/index.js [hunk 2]
```diff
@@ -8,84 +8,54 @@ module.exports = function (context, req) {
...
-        }
-
+    batch_client.pool.get(poolId).then((poolInfo) => {
+        context.log(`pool state: ${poolInfo.state}`);
+        return isAutoScaleEnabled(poolInfo);
+    }).then(poolInfo => {
+        context.log(`Auto Scale: ${poolInfo.enableAutoScale}`);
+        return ensureAutoScaleSet(batch_client, poolInfo);
+    }).then(_ => {
+        return evaluateAutoScale(batch_client, poolId, maxNodes);
```
Comment (noopkat) [inline EvaluateAutoScale/index.js:21]: this is purely stylistic, but can be rolled up to read:

```javascript
}).then(_ => evaluateAutoScale(batch_client, poolId, maxNodes))
  .then(...
```

File: EvaluateAutoScale/index.js [hunk 3]
```diff
@@ -8,84 +8,54 @@ module.exports = function (context, req) {
...
+function isAutoScaleEnabled(poolInfo){
+    if(poolInfo.state != "active")
+    {
+        Promise.reject({code: "notActive"});
+    }
+
+    return Promise.resolve(poolInfo);
+}
+
+function ensureAutoScaleSet(batch_client, poolInfo){
```
Comment (noopkat) [inline EvaluateAutoScale/index.js:43]: this function is 💯

File: EvaluateAutoScale/index.js [hunk 4]
```diff
@@ -8,84 +8,54 @@ module.exports = function (context, req) {
...
+    }).then(_ => {
+        return evaluateAutoScale(batch_client, poolId, maxNodes);
+    }).then(evalResult => {
+        context.log("Auto Scale Results:");
+        context.log(evalResult.results.replace("\$/gi", os.EOL + "\t$"));
+        context.done();
     }).catch(err => {
         context.log('An error occurred.');
-        context.log(err);
+        printErrors(context, err);
```
Comment (noopkat) [inline EvaluateAutoScale/index.js:28]: 👌

**Review by noopkat — COMMENTED** [review]
respect

File: EvaluateAutoScale/index.js (hunk 1 above)
Comment (jsturtevant) [reply EvaluateAutoScale/index.js]: woops yes!

**Comment by jsturtevant** [comment]
@noopkat Thanks for the advice and review!

**Pull request merged**